### PR TITLE
Add additional gcloud miscellaneous tooling

### PR DESCRIPTION
**Pull request by noursaidi**
Included more of my scripts including
- record permanent history of IoT Core configs
- save copy of raw message into GCS
- Save message log and telemetry into BQ
- save hardware and software from state messages into BQ
- log connection and disconnection events into BQ
- basic pubsub mirror script

Also moved deploy scripts into their own directory 

All have been tested/deployed in my projects, however here I have modified slightly to fix unnecesary cloud function logs and service account issues, but have not (at least at time of writing) tested the modified versions work. Also atm they also all use the app engine default credentials which might not have sufficient permissions in some projects, although creating a service account per cloud function is very quick to do.

### CODE REVIEW

File: misc/gcloud_iot_connection_log/README.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# GCloud IoT Connection Logs
+
+Stores all device connections and disconnection from IoT Core into BigQuery
+
+**NOTE**
+- Requires **Cloud Logging** on devices or registry be set to `INFO` or more.
+- IoT Core has a default log entries limit of 2000 per second. If a regsitry has `DEBUG` level logging, this may very quickly be exceeding, and will result in missing connection or disconnection log events
```
Comment (johnrandolph) [inline misc/gcloud_iot_connection_log/README.md:7]: nit: registry

File: misc/gcloud_pubsub_mirror/mirror.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+future = sub_client.subscribe(f"projects/{args.source_project}/subscriptions/{args.source_subscription}", subscribe_callback)
+print("Listening to pubsub, please wait ...")
+
+while True:
+    try:
+        future.result(timeout=5)
+    except futures.TimeoutError:
+        continue
+    except (futures.CancelledError, KeyboardInterrupt):
+        future.cancel()
```
Comment (johnrandolph) [inline misc/gcloud_pubsub_mirror/mirror.py:47]: do you want to fall through here?

File: misc/gcloud_store_message/deploy [hunk 3]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+if (( $# < 4 )); then
+    echo $0 PROJECT_ID DATASET_ID LOCATION TRIGGER_TOPIC [--drop]
```
Comment (johnrandolph) [inline misc/gcloud_store_message/deploy:4]: Does this script support --drop ? maybe remove it?

**Review by johnrandolph — COMMENTED** [review]
No comments on the .js

**Review by grafnu — APPROVED** [review]

**Pull request merged**